lib/LanguageServerWorseReflection/Workspace/WorkspaceIndex.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Phpactor\Extension\LanguageServerWorseReflection\Workspace;
+
+use Phpactor\TextDocument\TextDocument;
+use Phpactor\TextDocument\TextDocumentBuilder;
+use Phpactor\TextDocument\TextDocumentUri;
+use Phpactor\WorseReflection\Core\Name;
+use Phpactor\WorseReflection\Core\Reflector\SourceCodeReflector;
+use RuntimeException;
+
+class WorkspaceIndex
+{
+    /**
+     * @var SourceCodeReflector
+     */
+    private $reflector;
+
+    /**
+     * @var array<string, TextDocument>
+     */
+    private $byName = [];
+
+    /**
+     * @var array<string, TextDocument>
+     */
+    private $byFunction = [];
```

Comment:
renmove - although this is broken everywhere, functions and classes can share the same name.